addons/full-upgrade/run-upgrade.sh

```diff
@@ -0,0 +1,252 @@
+#!/bin/bash
+
+set -o nounset -o pipefail -o errexit
+
+# functions come from addons/functions but are packaged
+# inside full-upgrade directory to make full-upgrade package self-contained
+source /usr/local/pf/addons/full-upgrade/helpers.functions
+source /usr/local/pf/addons/full-upgrade/database.functions
+source /usr/local/pf/addons/full-upgrade/configuration.functions
+
+function backup_git_commit_id() {
+  if [ `cat /usr/local/pf/conf/git_commit_id` = "%{git_commit}" ]; then
+    echo "Broken git_commit_id detected, will have to guess which commit ID this build is based on"
+    
+    if [ `grep oauth2 conf/radiusd/packetfence-tunnel.example | wc -l` -eq 4 ]; then
+      echo "Detected that this build was made before 6f33f85713ec329b73572a073ee6419bab7f8c5e."
+      echo "Using the commit ID of v11.0.0 in conf/git_commit_id"
+      echo -n a0c25391e58310b1b2be937013a9808bfd7c1071 > /usr/local/pf/conf/git_commit_id
+    else
+      echo "Detected that this build was made using 6f33f85713ec329b73572a073ee6419bab7f8c5e or a later commit."
+      echo "Using the commit ID 6f33f85713ec329b73572a073ee6419bab7f8c5e in conf/git_commit_id"
+      echo -n 6f33f85713ec329b73572a073ee6419bab7f8c5e > /usr/local/pf/conf/git_commit_id
+    fi
+  fi
+  rm -f /usr/local/pf/conf/git_commit_id.preupgrade
+  cp -a /usr/local/pf/conf/git_commit_id{,.preupgrade}
+}
+
+function backup_pf_release() {
+  rm -f /usr/local/pf/conf/pf-release.preupgrade
+  cp -a /usr/local/pf/conf/pf-release{,.preupgrade}
+}
+
+function upgrade_packetfence_package() {
+  if is_rpm_based; then
+    yum_upgrade_packetfence_package $1
+  elif is_deb_based; then
+    apt_upgrade_packetfence_package $1
+  else
+    echo "Unable to detect package manager to upgrade PacketFence"
+    exit 1
+  fi
+}
+
+function find_latest_stable() {
+  OS=""
+  if is_rpm_based; then
+    OS="RHEL-8"
+  elif is_deb_based; then
+    OS="Debian-11"
+  fi
+  curl https://www.packetfence.org/downloads/PacketFence/latest-stable-$OS.txt
```

Comment:
If suggest to do a GitHub API call to get latest stable based on `stable` tag.
We decided to not use it anymore but it could be interested to use it for that purpose.

Reply:
The thing is the stable tag will differ for RHEL8 and RHEL9 when we get there so we'd need a stable-rhel8 stable-debian11, etc

By putting it in a file here, we're able to stop updating the stable release when we switch OS

Reply:
I understand your idea and your approach. I will see if I can automate update of these two files when we release.

Reply:
That would indeed be the goal, the release pipeline should be performing this, not one of us manually

Reply:
Should be ready now. I will run a pipeline and see if it works. Meanwhile, could you add support to pull `latest-devel-$OS-txt` when we are testing on `devel` or on development branches ?

Reply:
Pipeline triggered with `BUILD_ARTIFACTS_WEBSITE=yes`: 
https://gitlab.com/inverse-inc/packetfence/-/pipelines/382749074

It should create `latest-devel-$OS.txt` files.